src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzerTests.cs

```diff
@@ -102,14 +105,347 @@ End Class
+                    AdditionalFiles = { (".editorconfig", editorConfigText) },
+                    ExpectedDiagnostics = { GetCSharpCA1501ExpectedDiagnostic(1, 14, "MyUC", 7, 6, "UserControl, ContainerControl, ScrollableControl, Control, Component, MarshalByRefObject, Object") },
+                },
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithWinForms,
+            }.RunAsync();
+
+            await new VerifyVB.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+Public Class MyUC
+    Inherits System.Windows.Forms.UserControl
+End Class"
+                    },
+                    AdditionalFiles = { (".editorconfig", editorConfigText) },
+                    ExpectedDiagnostics = { GetBasicCA1501ExpectedDiagnostic(2, 14, "MyUC", 7, 6, "UserControl, ContainerControl, ScrollableControl, Control, Component, MarshalByRefObject, Object") },
+                },
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithWinForms,
+            }.RunAsync();
+        }
+
+        [Fact, WorkItem(1839, "https://github.com/dotnet/roslyn-analyzers/issues/1839")]
+        public async Task CA1501_WildcardTypePrefixNoNamespace()
+        {
+            var editorConfigText = "dotnet_code_quality.CA1501.inheritance_excluded_type_names = T:SomeClass*";
+
+            var codeMetricsConfigText = @"
+# FORMAT:
+# 'RuleId'(Optional 'SymbolKind'): 'Threshold'
+
+CA1501: 1
+";
+
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+namespace MyCompany.MyProduct.MyFunction
+{
+    public class SomeClass {}
+    public class C1 : SomeClass {}
+
+    public class SomeClass1 {}
+    public class SomeClass2 : SomeClass1 {}
+    public class C2 : SomeClass2 {}
+}
+
+public class SomeClass {}
+public class C1 : SomeClass {}
+
+public class SomeClass1 {}
+public class SomeClass2 : SomeClass1 {}
+public class C2 : SomeClass2 {}"
+                    },
+                    AdditionalFiles =
+                    {
+                        (".editorconfig", editorConfigText),
+                        (AdditionalFileName, codeMetricsConfigText),
+                    },
+                    ExpectedDiagnostics =
+                    {
+                        GetCSharpCA1501ExpectedDiagnostic(5, 18, "C1", 2, 2, "SomeClass, Object"),
+                        GetCSharpCA1501ExpectedDiagnostic(8, 18, "SomeClass2", 2, 2, "SomeClass1, Object"),
+                        GetCSharpCA1501ExpectedDiagnostic(9, 18, "C2", 3, 2, "SomeClass2, SomeClass1, Object"),
+                    }
+                },
+            }.RunAsync();
+        }
+
+        [Fact, WorkItem(1839, "https://github.com/dotnet/roslyn-analyzers/issues/1839")]
+        public async Task CA1501_WildcardTypePrefixWithNamespace()
```

Comment:
Can this test be combined with the previous test and use InlineData and Theory attributes for different editorconfig specifications?